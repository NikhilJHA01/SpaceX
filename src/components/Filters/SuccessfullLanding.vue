<template>
  <el-checkbox-group
    v-model="landingSucessArray"
    size="small"
    text-color="#000000"
    fill="#5daf34"
    @change="changeFilter()"
  >
    <el-checkbox-button v-for="value in options" :label="value" :key="value">{{
      value
    }}</el-checkbox-button>
  </el-checkbox-group>
</template>

<script>
import { BooleanArray } from "../../utility/constant.js";

export default {
  name: "SuccessfullLanding",
  data() {
    return {
      landingSucessArray: [],
      options: BooleanArray
    };
  },
  methods: {
    changeFilter() {
      this.$store.dispatch(
        "Set_Filter_By_Landing_Success",
        this.landingSucessArray
      );
      this.$emit("changeFilter");
    }
  }
};
</script>
