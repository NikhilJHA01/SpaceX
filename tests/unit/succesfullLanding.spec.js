import { mount } from "@vue/test-utils";
import SuccessfullLanding from "@/components/Filters/SuccessfullLanding";

describe("SuccessfullLanding Component", () => {
  const wrapper = mount(SuccessfullLanding, {
    data() {
      return {
        landingSucessArray: [],
        options: ["TRUE", "FALSE"]
      };
    }
  });
  it("renders the SuccessfullLanding", async () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has a checkbox-group", () => {
    const checkbox_group = wrapper.find("el-checkbox-group");
    expect(checkbox_group.exists()).toBe(true);
  });

  it("Filter is not selected by default", () => {
    expect(wrapper.vm.$data.landingSucessArray).toHaveLength(0);
  });
});
