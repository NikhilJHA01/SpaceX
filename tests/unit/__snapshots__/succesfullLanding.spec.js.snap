// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessfullLanding Component renders the SuccessfullLanding 1`] = `
<el-checkbox-group size="small" text-color="#000000" fill="#5daf34">
  <el-checkbox-button label="TRUE">TRUE</el-checkbox-button>
  <el-checkbox-button label="FALSE">FALSE</el-checkbox-button>
</el-checkbox-group>
`;
